src/Functions/parseTimeDelta.cpp

```diff
@@ -50,8 +51,26 @@ namespace
         {"second", 1},
         {"sec", 1},
         {"s", 1},
+
+        {"milliseconds", 1e-3},
+        {"millisecond", 1e-3},
+        {"millisec", 1e-3},
+        {"ms", 1e-3},
+
+        {"microseconds", 1e-6},
+        {"microsecond", 1e-6},
+        {"microsec", 1e-6},
+        {"μs", 1e-6},
+        {"us", 1e-6},
+
+        {"nanoseconds", 1e-9},
+        {"nanosecond", 1e-9},
+        {"nanosec", 1e-9},
+        {"ns", 1e-9},
     };
 
+    const std::unordered_set<char> separators = {' ', ';', '-', '+', ',', ':'};
```

Comment:
It can be slower than a bunch of comparisons.

Reply:
Fixed